coderd/database/dbfake/dbfake.go

```diff
@@ -2791,18 +2791,69 @@ func (q *FakeQuerier) GetUsersByIDs(_ context.Context, ids []uuid.UUID) ([]datab
 	return users, nil
 }
 
-func (q *FakeQuerier) GetWorkspaceAgentByAuthToken(_ context.Context, authToken uuid.UUID) (database.WorkspaceAgent, error) {
+func (q *FakeQuerier) GetWorkspaceAgentAndOwnerByAuthToken(_ context.Context, authToken uuid.UUID) (database.GetWorkspaceAgentAndOwnerByAuthTokenRow, error) {
 	q.mutex.RLock()
 	defer q.mutex.RUnlock()
-
-	// The schema sorts this by created at, so we iterate the array backwards.
-	for i := len(q.workspaceAgents) - 1; i >= 0; i-- {
-		agent := q.workspaceAgents[i]
-		if agent.AuthToken == authToken {
-			return agent, nil
+	var rows []database.GetWorkspaceAgentAndOwnerByAuthTokenRow
+AgentLoop:
+	for _, agt := range q.workspaceAgents {
+		if agt.AuthToken != authToken {
+			continue AgentLoop
+		}
+		// get the related workspace and user
+	ResourceLoop:
+		for _, res := range q.workspaceResources {
+			if agt.ResourceID != res.ID {
+				continue ResourceLoop
+			}
+		BuildLoop:
+			for _, build := range q.workspaceBuilds {
+				if build.JobID != res.JobID {
+					continue BuildLoop
+				}
+			WorkspaceLoop:
+				for _, ws := range q.workspaces {
+					if build.WorkspaceID != ws.ID {
+						continue WorkspaceLoop
```

Comment:
None of these tags seem required, and plain continue would operate on the correct loop. Should we simplify, or could it be a bug?

Reply:
The tags were only really necessary in a previous implementation, we can remove them.